htdocs/supplier_proposal/class/supplier_proposal.class.php

```diff
@@ -1403,8 +1405,10 @@ public function valid($user, $notrigger = 0)
 			$soc = new Societe($this->db);
 			$soc->fetch($this->socid);
 
+			if ($soc < 0) return -1;
```

Comment:
$soc is an object, it can' be compared to < 0 
May be you want to do 
```
$result = $soc->fetch($this->socid);
if ($result < 0) return -1;
```

Reply:
Oh... Yes, of course !